homeassistant/components/homekit_controller/__init__.py

```diff
@@ -304,12 +304,13 @@ def _setup_characteristic(self, char):
     @property
     def unique_id(self):
         """Return the ID of this device."""
-        return self._address
+        serial = self._accessory_info.get('serial-number', 'unknown')
```

Comment:
Will the unique_id still be unique if serial is unknown?

Reply:
Answered in the PR description.

Reply:
Or maybe not? Feel free to comment here if it needs clarification.

Reply:
Looking at the spec, technically every accessory must expose a single instance of the accessory-information service and every one of those must have a single serial-number characteristic (it is a required attribute), so no certified accessory will not have this field. So I guess i should change it to `self._accessory_info['serial-number']`.